packages/@react-aria/menu/src/useSubmenuTrigger.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaMenuItemProps} from './useMenuItem';
+import {AriaMenuOptions} from './useMenu';
+import type {AriaPopoverProps} from '@react-aria/overlays';
+import {FocusableElement, FocusStrategy, KeyboardEvent, PressEvent, Node as RSNode} from '@react-types/shared';
+import {RefObject, useCallback, useRef} from 'react';
+import type {SubmenuTriggerState} from '@react-stately/menu';
+import {useEffectEvent, useId, useLayoutEffect} from '@react-aria/utils';
+import {useLocale} from '@react-aria/i18n';
+import {useSafelyMouseToSubmenu} from './useSafelyMouseToSubmenu';
+
+export interface AriaSubmenuTriggerProps {
+  /** An object representing the submenu trigger menu item. Contains all the relevant information that makes up the menu item. */
+  node: RSNode<unknown>,
+  /** Whether the submenu trigger is disabled. */
+  isDisabled?: boolean,
+  /** Type of the submenu being rendered. */
+  submenuType?: 'dialog' | 'menu',
+  /** Ref of the menu that contains the submenu trigger. */
+  parentMenuRef: RefObject<HTMLDivElement>,
+  /** Ref of the submenu opened by the submenu trigger. */
+  submenuRef: RefObject<HTMLDivElement>
+}
+
+interface SubmenuTriggerProps extends AriaMenuItemProps {
+  /** Whether the submenu trigger is in an expanded state. */
+  isOpen: boolean
+}
+
+interface SubmenuProps<T> extends AriaMenuOptions<T> {
+  /** The level of the submenu. */
+  submenuLevel: number
+}
+
+export interface SubmenuTriggerAria<T> {
+  /** Props for the submenu trigger menu item. */
+  submenuTriggerProps: SubmenuTriggerProps,
+  /** Props for the submenu controlled by the submenu trigger menu item. */
+  submenuProps: SubmenuProps<T>,
+  /** Props for the submenu's popover container. */
+  popoverProps: Pick<AriaPopoverProps, 'isNonModal'>,
```

Comment:
I only wanted the `isNonModal` prop definition from `AriaPopoverProps` since it has a couple required props that we don't provide from this hook. I could instead just redefine `isNonModal` here if we wanna get rid of the `Pick`, but figured it was the best way to maintain prop definition parity